### PR TITLE
[DELTA-OSS-EXTERNAL]Updated example Streaming

### PR DESCRIPTION
**Pull request by echang0929**
1. Added some println as comments for different sections
2. Added variable checkpointPath to replace hardcode in option
3. Changed duration of stream3 from 10000 to 20000 so as to get obvious 
results for updated Delta Table
4. Added a delete statement of checkpoint directory in the end

### CODE REVIEW

**Comment by databricks-cla-assistant** [comment]
[CLA assistant check](https://cla.dev.databricks.com/delta-io/delta?pullRequest=270) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.databricks.com/delta-io/delta?pullRequest=270) before we can accept your contribution.<br/><hr/>**zyibj** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dev.databricks.com/check/delta-io/delta?pullRequest=270) it.</sub>

File: examples/scala/src/main/scala/example/Streaming.scala [hunk 1]
```diff
@@ -36,6 +36,7 @@ object Streaming {
 
     import spark.implicits._
 
+    println("=== Section 1: write and read delta table normally, and for Section 3")
```
Comment (mukulmurthy) [inline examples/scala/src/main/scala/example/Streaming.scala]: I'd replace "normally" with "using batch queries" and the last bit to "initialize table for later sections".

File: examples/scala/src/main/scala/example/Streaming.scala [hunk 2]
```diff
@@ -101,15 +107,18 @@ object Streaming {
...
 
     println("Delta Table after streaming upsert")
     deltaTable.toDF.show()
 
+
+    println("=== In the end, clean all paths")
     // Cleanup
     FileUtils.deleteDirectory(new File(path))
     FileUtils.deleteDirectory(new File(tablePath2))
+    FileUtils.deleteDirectory(new File(checkpointPath))
```
Comment (mukulmurthy) [inline examples/scala/src/main/scala/example/Streaming.scala]: Thank you for fixing this!

File: examples/scala/src/main/scala/example/Streaming.scala [hunk 3]
```diff
@@ -70,6 +74,8 @@ object Streaming {
     stream2.awaitTermination(10000)
     stream2.stop()
 
+
+    println("=== Section 3: Streaming upgrades in update mode")
```
Comment (mukulmurthy) [inline examples/scala/src/main/scala/example/Streaming.scala]: I'd phrase this as "Streaming upserts using MERGE".

File: examples/scala/src/main/scala/example/Streaming.scala [hunk 4]
```diff
@@ -70,6 +74,8 @@ object Streaming {
...
     // Function to upsert microBatchOutputDF into Delta Lake table using merge
     println("Streaming upgrades in update mode")
```
Comment (mukulmurthy) [inline examples/scala/src/main/scala/example/Streaming.scala]: this line can be deleted now.

File: examples/scala/src/main/scala/example/Streaming.scala [hunk 5]
```diff
@@ -45,14 +46,17 @@ object Streaming {
     val df = spark.read.format("delta").load(path)
     df.show()
 
+
+    println("=== Section 2: write and read delta stream")
     println("Streaming write")
```
Comment (mukulmurthy) [inline examples/scala/src/main/scala/example/Streaming.scala]: stream -> "using structured streaming", and we can get rid of the old println that was there before.

File: examples/scala/src/main/scala/example/Streaming.scala [hunk 6]
```diff
@@ -45,14 +46,17 @@ object Streaming {
...
+    val checkpointPath = new File("/tmp/checkpoint").getCanonicalPath
     val stream = streamingDf
       .select($"value" as "id")
       .writeStream
       .format("delta")
-      .option("checkpointLocation", new File("/tmp/checkpoint").getCanonicalPath)
+      .option("checkpointLocation", checkpointPath)
       .start(tablePath2)
 
     stream.awaitTermination(10000)
```
Comment (mukulmurthy) [inline examples/scala/src/main/scala/example/Streaming.scala]: Can remove the println just below this now.

File: examples/scala/src/main/scala/example/Streaming.scala (hunk 1 above)
Comment (echang0929) [reply examples/scala/src/main/scala/example/Streaming.scala]: Yes, I updated with your comment.

File: examples/scala/src/main/scala/example/Streaming.scala (hunk 3 above)
Comment (echang0929) [reply examples/scala/src/main/scala/example/Streaming.scala]: Yes, I updated with your comment.

File: examples/scala/src/main/scala/example/Streaming.scala (hunk 4 above)
Comment (echang0929) [reply examples/scala/src/main/scala/example/Streaming.scala]: Yes, I deleted line 80

File: examples/scala/src/main/scala/example/Streaming.scala (hunk 5 above)
Comment (echang0929) [reply examples/scala/src/main/scala/example/Streaming.scala]: Yes, I updated with your comment.

File: examples/scala/src/main/scala/example/Streaming.scala (hunk 6 above)
Comment (echang0929) [reply examples/scala/src/main/scala/example/Streaming.scala]: Yes, I removed the line println("Reading from stream")

**Comment by echang0929** [comment]
Thank you very much for your review and suggestions. Here are logs after updating:
```
=== Section 1: write and read delta table using batch queries, and initialize table for later sections
+---+
| id|
+---+
|  4|
|  3|
|  2|
|  0|
|  1|
+---+

=== Section 2: write and read delta using structured streaming
-------------------------------------------
Batch: 0
-------------------------------------------
+---+
| id|
+---+
|  0|
|  1|
|  2|
|  3|
|  4|
|  5|
|  6|
|  7|
+---+

=== Section 3: Streaming upserts using MERGE
Original Delta Table
+---+
| id|
+---+
|  4|
|  3|
|  2|
|  0|
|  1|
+---+

Delta Table after streaming upsert
+---+
| id|
+---+
|  1|
| 11|
| 10|
|  5|
| 12|
|  2|
|  3|
|  6|
| 14|
|  9|
|  4|
| 13|
|  7|
|  8|
|  0|
+---+

=== In the end, clean all paths
```

**Comment by rahulsmahadev** [comment]
@echang0929 Thank you for this contribution, however we would want to temporarily hold off from merging this pull request as we are making the 0.5.0 release this week and we are updating the examples for that. It would be great if you could rebase your PR once 0.5.0 is released. We will merge it then.

Thanks

**Comment by echang0929** [comment]
Sure, thank you both.

**Comment by xianyinxin** [comment]
Hi @echang0929 , would you pls describe the objective of this PR? And what 'DELTA-OSS-EXTERNAL' means？Sorry for this but i'm not very clear about the background.

**Comment by xianyinxin** [comment]
@mukulmurthy got it, thanks for the explanation. :)

**Pull request closed without merging**